esda/moran.py

```diff
@@ -1718,4 +1726,62 @@ def neighbors_perm(
         if keep:
             out[i,] = rstats
         accumulator[i] = numpy.sum(rstats >= observed[i])
-    return accumulator, out
\ No newline at end of file
+    return accumulator, out
+
+@njit(fastmath=True)
+def neighbors_perm_plus(
+    z: numpy.ndarray,
+    observed: numpy.ndarray,
+    cardinalities: numpy.ndarray,
+    weights: numpy.ndarray,
+    permutations: int,
+    keep: bool,
+):
+    z = z.copy().reshape(-1, 1)
+    n = len(z)
+    larger = numpy.zeros((n,), dtype=numpy.int64)
+    if keep:
+        rlisas = numpy.empty((n, permutations))
+    else:
+        rlisas = numpy.empty((1, 1))
+
+    max_card = cardinalities.max()
+    permutations = vec_permutations(max_card, permutations)
+    mask = numpy.ones((n,), dtype=numpy.int8) == 1
+    wloc = 0
+    for i in prange(n):
+        cardinality = cardinalities[i]
+        ### this chomps the first `cardinality` weights off of `weights`
+        weights_i = weights[wloc : (wloc + cardinality)]
+        wloc += cardinality
+        zi = z[i]
+        mask[i] = False
+        z_no_i = z[
```

Comment:
```python
z_no_i = z[mask, ]
numpy.random.shuffle(z_no_i)
```
does this randomize the source z? or just z_no_i? 

if it does, we can still do
```python
z_no_i = numpy.delete(z, i) # copies, so remove the z.copy() at the start
numpy.shuffle(z_no_i)
```